pkg/service/FluxService/helper.go

```diff
@@ -0,0 +1,363 @@
+		Name:                  appDetail.Name,
+		HealthStatus:          appDetail.HealthStatus,
+		SyncStatus:            appDetail.SyncStatus,
+		FluxAppDeploymentType: string(appDetail.FluxAppDeploymentType),
+		EnvironmentDetail: &client.EnvironmentDetails{
+			ClusterName: appDetail.EnvironmentDetails.ClusterName,
+			ClusterId:   int32(appDetail.EnvironmentDetails.ClusterId),
+			Namespace:   appDetail.EnvironmentDetails.Namespace,
+		},
+	}
+}
+func getFluxSpecKubeConfig(obj map[string]interface{}) (bool, error) {
+	spec, found, err := unstructured.NestedMap(obj, "spec")
+	if err != nil {
+		return false, err
+	}
+	if !found {
+		return false, nil
+	}
+	_, found, err = unstructured.NestedFieldCopy(spec, "kubeConfig")
+	if err != nil {
+		return false, err
+	}
+	if !found {
+		return false, nil
+	}
+	return true, nil
+}
+func parseObjMetadata(s string) (FluxKsResourceDetail, error) {
+	index := strings.Index(s, FieldSeparator)
+	if index == -1 {
+		return NilObjMetadata, fmt.Errorf("unable to parse stored object metadata: %s", s)
+	}
+	namespace := s[:index]
+	s = s[index+1:]
+	// Next, parse last field kind
+	index = strings.LastIndex(s, FieldSeparator)
+	if index == -1 {
+		return NilObjMetadata, fmt.Errorf("unable to parse stored object metadata: %s", s)
+	}
+	kind := s[index+1:]
+	s = s[:index]
+	// Next, parse next to last field group
+	index = strings.LastIndex(s, FieldSeparator)
+	if index == -1 {
+		return NilObjMetadata, fmt.Errorf("unable to parse stored object metadata: %s", s)
+	}
+	group := s[index+1:]
+	// Finally, second field name. Name may contain colon transcoded as double underscore.
+	name := s[:index]
+	name = strings.ReplaceAll(name, ColonTranscoded, ":")
+	// Check that there are no extra fields by search for fieldSeparator.
+	if strings.Contains(name, FieldSeparator) {
+		return NilObjMetadata, fmt.Errorf("too many fields within: %s", s)
+	}
+	// Create the ObjMetadata object from the four parsed fields.
+	id := FluxKsResourceDetail{
+		Namespace: namespace,
+		Name:      name,
+		Group:     group,
+		Kind:      kind,
+	}
+	return id, nil
+}
+func inventoryExists(obj map[string]interface{}) bool {
+	statusObj, found, err := unstructured.NestedMap(obj, STATUS)
+	if err != nil || !found {
+		return false
+	}
+	_, found, err = unstructured.NestedMap(statusObj, INVENTORY)
+	if err != nil || !found {
+		return false
+	}
+	return true
+}
+func fetchInventoryList(obj map[string]interface{}) ([]FluxKsResourceDetail, error) {
+	var inventoryResources []FluxKsResourceDetail
+	childResourcesMap, err := getInventoryMap(obj)
+	if err != nil {
+		return nil, err
+	}
+	for childResourceId, version := range childResourcesMap {
+		fluxResource, err := parseObjMetadata(childResourceId)
+		if err != nil {
+			err = fmt.Errorf("unable to parse stored object metadata: %s", childResourceId)
+			return nil, err
+		}
+		fluxResource.Version = version
+		inventoryResources = append(inventoryResources, fluxResource)
```

Comment:
this section might panic, initialise inventoryResources using make and also put len for make